src/trio/_core/_tests/test_run.py

```diff
@@ -1088,21 +1095,28 @@ async def child() -> None:
 
         try:
             raise KeyError
-        except Exception:
-            with suppress(Exception):
+        except KeyError:
+            with suppress(ValueError):
                 await sleep_forever()
             raise
 
-    with pytest.raises(KeyError):
+    with pytest.raises(KeyError) as excinfo:
         async with _core.open_nursery() as nursery:
             nursery.start_soon(child)
             await wait_all_tasks_blocked()
             _core.reschedule(not_none(child_task), outcome.Error(ValueError()))
 
+    assert excinfo.value.__context__ is None
+
 
-# Similar to previous test -- if the ValueError() gets sent in via 'throw',
-# then Python's normal implicit chaining stuff is broken.
-async def test_exception_chaining_after_yield_error() -> None:
+# Similar to previous test -- if the `ValueError()` gets sent in via 'throw' and
+# propagates out, then CPython doesn't set its `__context__` so normal implicit
+# exception chaining is broken:
+#   https://bugs.python.org/issue25612 (Example 2)
+#   https://bugs.python.org/issue25683
+#   https://bugs.python.org/issue29587 (Example 1)
+# This is fixed in CPython >= 3.9.
```

Comment:
I'm slightly confused that the test doesn't have any calls to `sys.version_info` when it talks about differing behaviour on different python versions. Can maybe clarify the comment a bit more in that respect?

Reply:
Right. It's only if we applied that patch that we'd get different behaviour depending on python version. And the workaround implemented by trio, for this specific bug/test, is redundant on py3.9+, and when we drop support for 3.8 in oct 2024 we *could* remove this test and any corresponding workaround.